### DISCUSSION TITLE
How to print a space after 'username' module only if there is no hostname present?

### DISCUSSION THREAD

**Discussion by ftpd** [starship/starship#5267] [Q&A]
I have the following segments:

```
format="""
$python\
$username\
$hostname\
$directory\
$git_branch\
$git_status\
$jobs\
$character
"""
```

and setup for (some of) them:

```
[username]
format = '[$user]($style)'
style_user = 'normal'
style_root = 'bold 196'

[hostname]
format = '([@$hostname]($style) )'
style = 'normal'

[directory]
format = '[($read_only )]($read_only_style)[$path]($style) '
style = '37'
read_only_style = '37'
read_only = ''
truncation_length = 0
```

and on remote hosts it looks as intended, so `user@host ~ $`. The problem is with root on my local machine - the username shows (which I like and I want to have), but because the hostname is hidden (also - as intended), there is no space before 'username' and 'directory' module, so it becomes `root~ #`.

Can I somehow force styling the 'username' module with a space, but only if there is no 'hostname' present? Obviously I could change the `format` of 'username' module to be `[$user]($style) `, so with space, but it will then look awful on a remote hosts (like `user @host ~ #`).

**Comment by craighurley** [comment]
How about moving the space around like this:

```
[username]
format = '[$user]($style)'

[hostname]
format = '([@$hostname]($style))'

[directory]
format = ' [($read_only )]($read_only_style)[$path]($style) '
```

**Reply by ftpd** [reply]
It will then print space for the prompt of the non-root local account (so, actually, the prompt I'm  seeing for the most of the time).
